extensions/lucide-icons/src/index.tsx

```diff
@@ -1,45 +1,101 @@
-import { Action, ActionPanel, Color, environment, Grid, Icon } from "@raycast/api";
-import { useCachedState } from "@raycast/utils";
-import { readdirSync, readFileSync } from "fs";
-import { join } from "path";
-import { useMemo } from "react";
+import { Action, ActionPanel, Color, Grid, Icon } from "@raycast/api";
+import { useCachedPromise, useCachedState } from "@raycast/utils";
+import fetch from "node-fetch";
 
 // Lucide's toPascalCase function
 export const toPascalCase = (string: string) => {
-  const camelCase = string.replace(/^([A-Z])|[\s-_]+(\w)/g, (match, p1, p2) =>
-    p2 ? p2.toUpperCase() : p1.toLowerCase()
+  const camelCase = string.replace(/^([A-Z])|[\s-_]+(\w)/g, (_match, p1, p2) =>
+    p2 ? p2.toUpperCase() : p1.toLowerCase(),
   );
 
   return camelCase.charAt(0).toUpperCase() + camelCase.slice(1);
 };
 
+type LucideIcon = {
+  name: string;
+  content: string;
+  path: string;
+  component: string;
+};
+
 export default function Command() {
-  const files = useMemo(
-    () =>
-      readdirSync(join(environment.assetsPath, "icons"))
-        .filter((file) => file.endsWith(".svg"))
-        .map((file) => {
-          return [join(environment.assetsPath, "icons", file), file.split(".")[0]];
-        }),
-    []
+  const { data, isLoading } = useCachedPromise(
+    async (iconNodesUrl: string) => {
+      const response = await fetch(iconNodesUrl)
+        .then((res) => res.json() as Promise<Record<string, unknown>>)
+        .then((data) => Object.keys(data));
+
+      const promises = response.map(async (iconName) => {
+        const url = `https://lucide.dev/api/icons/${iconName}`;
+        const content = await fetch(url).then((res) => res.text());
+
+        return {
+          name: iconName,
+          content,
+          path: url,
+          component: `<${toPascalCase(iconName)} />`,
+        };
+      });
+
+      const icons = await Promise.all(promises);
+      setIcons(icons);
+    },
+    ["https://lucide.dev/api/icon-nodes"],
+    {
+      keepPreviousData: true,
+    },
   );
 
+  const [icons, setIcons] = useCachedState<LucideIcon[] | null>("icons", null);
+
+  console.log(data);
+  // const files = useMemo(
+  //   () =>
+  //     readdirSync(join(environment.assetsPath, "icons"))
+  //       .filter((file) => file.endsWith(".svg"))
+  //       .map((file) => {
+  //         return [join(environment.assetsPath, "icons", file), file.split(".")[0]];
+  //       }),
+  //   []
+  // );
+
   const [color, setColor] = useCachedState<Color>("color", Color.PrimaryText);
   const [columns, setColumns] = useCachedState<number>("size", 8);
 
   return (
-    <Grid columns={columns} inset={Grid.Inset.Large}>
-      {files.map(([path, name]) => (
+    <Grid columns={columns} inset={Grid.Inset.Large} isLoading={isLoading || !data}>
```

Comment:
```suggestion
    <Grid columns={columns} inset={Grid.Inset.Large} isLoading={isLoading}>
```

Reply:
And even maybe an EmptyView that shows something while loading is true?

Reply:
Will keep an eye on that